userspace/libsinsp/plugin.cpp

```diff
@@ -610,6 +620,75 @@ bool sinsp_plugin::resolve_dylib_symbols(std::string& errstr) {
 	return true;
 }
 
+bool sinsp_plugin::check_required_schema_version(std::string& err) {
+	// Check if the plugin consumes driver events and needs schema version validation
+	bool needs_schema_check = false;
+
+	// Check if plugin provides parsing capabilities and consumes syscall events
+	if(m_caps & CAP_PARSING) {
+		for(const auto& event_type : m_parse_event_codes) {
+			if(event_type != ppm_event_code::PPME_PLUGINEVENT_E) {
+				needs_schema_check = true;
+				break;
+			}
+		}
+	}
+
+	// Check if plugin provides field extraction capabilities and consumes syscall events
+	if(!needs_schema_check && (m_caps & CAP_EXTRACTION)) {
+		for(const auto& event_type : m_extract_event_codes) {
+			if(event_type != ppm_event_code::PPME_PLUGINEVENT_E) {
+				needs_schema_check = true;
+				break;
+			}
+		}
+	}
+
+	// If schema check is needed, validate the schema version
+	if(needs_schema_check) {
+		// Default to schema version 3.0.0, which is the major schema version of the previous
+		uint32_t major = 3, minor = 0, patch = 0;
+		const char* required_version = NULL;
+		if(m_handle->api.get_required_schema_version != NULL) {
+			required_version = m_handle->api.get_required_schema_version(m_state);
+			if(required_version != NULL) {
+				if(sscanf(required_version,
+				          "%" PRIu32 ".%" PRIu32 ".%" PRIu32,
+				          &major,
+				          &minor,
+				          &patch) != 3) {
+					err = "plugin provided an invalid required schema version: '" +
+					      std::string(required_version) + "'";
+					return false;
+				}
+			}
+		}
+
+		uint32_t current_major = PPM_API_VERSION_MAJOR(SCAP_MINIMUM_DRIVER_SCHEMA_VERSION);
+		uint32_t current_minor = PPM_API_VERSION_MINOR(SCAP_MINIMUM_DRIVER_SCHEMA_VERSION);
+		uint32_t current_patch = PPM_API_VERSION_PATCH(SCAP_MINIMUM_DRIVER_SCHEMA_VERSION);
+		std::string failmsg("");
+		/* The plugin requires a minimum schema version */
+		if(current_major != major) {
+			failmsg = "major versions disagree";
+		} else if(current_minor < minor) {
+			failmsg = "minor schema version is less than the requested one";
+		}
+		/* Patch version differences are allowed (backward compatible) */
```

Comment:
I'd also check the patch, according to the Semver spec. 
The reason is simple: if the plugin requires a minimum version with a specific patch version, it means the plugin needs that patch.

This is also the common expected behavior.

Reply:
yes, I agree that as the API allows specifying it, it makes more sense to use it than silently ignore it